Sources/Caching/Caching.swift

```diff
@@ -0,0 +1,44 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  Caching.swift
+//
+//  Created by Jacob Zivan Rakidzich on 8/13/25.
+
+import Foundation
+
+/// An inteface representing a simple cache
+@_spi(Internal) public protocol Caching {
+
+    /// A URL for a cache directory if one is present
+    var cacheDirectory: URL? { get }
+
+    /// Store data to a url
+    func saveData(_ data: Data, to url: URL) throws
+
+    /// Check if there is content cached at the given path
+    func cachedContentExists(at path: String) -> Bool
```

Comment:
Should `path` be a URL instead of a String to be consistent with the param type in the `saveData` function?